### ISSUE TITLE
Plaid UI no longer opens

### ISSUE THREAD

**Issue by Isoaxe** [plaid/react-plaid-link#249]
I had this working previously, but it no longer does and I don't think that the code has been modified in the interim. All that happened relating to Plaid was being granted access to the development environment.

For the `const { open, ready } = usePlaidLink({ token, onSuccess })` function, the `token` is successfully retrieved. `ready` also evaluates to `true`. The `open()` function is run, but the `onSuccess` callback does not execute. Instead, I get a `400` error as a `POST` request is made to `https://sandbox.plaid.com/link/workflow/next`. No other data is provided with that response.

Any ideas as to what the issue could be?

**Comment by Isoaxe** [comment]
Some further info: I'm building in React and am running `open()` from a `useEffect` hook when `ready` (and another flag). Was testing with a button click instead and that works (the Plaid UI gets displayed). So it seems to be an issue with the `useEffect` hook specifically. I've not changed the code here since it was working though.

**Comment by Isoaxe** [comment]
I actually just resolved it there. There was a modal that `<PlaidLink/>` was loaded from that closed, possibly before `open()` had finished running? I stopped the modal from closing and it's working fine now. No idea why it was working before and suddenly not now though, given that there were no code changes.

**Comment by connorskees** [comment]
Hello,

We also ran into this issue recently. The most likely explanation that we can come up with for why this started to silently fail is a silent change in the [script that is loaded by this library](https://cdn.plaid.com/link/v2/stable/link-initialize.js).

I am of the opinion that this issue should be re-opened. This is a _silent, breaking_ change and caused us to be unable to link bank accounts using Plaid for multiple days.

**Comment by skylarmb** [comment]
to clarify @Isoaxe you should not expect `onSuccess` to fire right after you call `open()`. `onSuccess` only fires after your user completes the entire Link flow and clicks "Continue" to return to your app. 

If you are seeing a 400 response somewhere in your console, that is an indication there was an error somewhere in the Link flow. I would highly recommend binding the [`onEvent`](https://plaid.com/docs/link/web/#onevent) function and [`onExit`](https://plaid.com/docs/link/web/#onexit) function from the hook so your application can be informed when errors happen within link. Right now you are essentially doing zero error handling so it is hard for you to catch those errors and report the request IDs and link session IDs associated with those errors to the [Plaid support team](https://dashboard.plaid.com/support). 

Example:  https://github.com/plaid/react-plaid-link/blob/master/examples/hooks.tsx#L26-L47

**Comment by Isoaxe** [comment]
Thanks for the error handling advice @skylarmb. Yes I realized that `onSuccess` executes at the end of the Plaid UI flow after I got it working.

Some of the relevant code you asked for:

From `EmailModal.js`:
```
async function continueToPayments() {
    setIsLoading(true);
    const { stripe_uid } = await createCustomer(email);
    const { client_secret, payment_intent_id } = await createSubscription(
      priceId,
      stripe_uid
    );
    setClientSecret(client_secret);
    setStripeUid(stripe_uid);
    setPaymentIntentId(payment_intent_id);
    setIsLoading(false);
    if (paymentMethod === "card") {
      setEmailModalVisible(false);
      setPaymentsModalVisible(true);
    } else if (paymentMethod === "ach") {
      setAchPayments(true);
      setEmailModalVisible(false);
    }
  }
```

In the above, `achPayments` is a flag that is used to trigger the Plaid UI flow in `PlaidLink.js`.

From `PlaidLink.js`:
```
const { open, ready } = usePlaidLink({
    token,
    onSuccess,
  });

```
```
useEffect(() => {
    if (achPayments && ready) {
      open();
      setAchPayments(false);
    }
  }, [achPayments, setAchPayments, ready, open]);
```
As mentioned in my previous closing comment, the issue was resolved by not switching off modal visibility after Plaid initilisation (i.e. Simply removing `setEmailModalVisible(false)` from `EmailModal.js`).

**Comment by Isoaxe** [comment]
I have actually discontinued Plaid from the app I'm working on at the client's request in favour of Stripe ACH direct debit that only very recently supports instant verification. So just pulled the above from my git history and won't be developing any further.

Don't keep this issue open on my behalf. I really only reopened it at @connorskees request.

**Comment by skylarmb** [comment]
> resolved by not switching off modal visibility after Plaid initilisation

I think there are two possible causes here

1) Most likely: Your email modal is conflicting with the Plaid modal because they are both modifying the global styles on the html/body in order to overlay themselves on the page. `setEmailModalVisible(false)` possibly also removed some styling of Plaid modal or in some way removed it from the page (e.g. was Plaid rendering from inside the modal and then it was closed, causing it to be unmounted?)

2) Less likely: is it possible that updating your state, mainly calling `setEmailModalVisible` after you open Plaid caused you to fire the `useEffect` that calls `open()` multiple times. We did recently discover an issue where calling `open()` repeatedly in a short period of time can cause Link to throw an error. When looking at this error we found that only a tiny handfull of developers had ever encountered the issue, but maybe you were one of those.


Either way it sounds like the issue is no longer relevant as you are not using Link anymore, so im closing this issue. Thanks for the details code samples!
